pkg/crc/api/handlers.go

```diff
@@ -213,22 +214,37 @@ func (h *Handler) GetConfig(c *context) error {
 
 	if len(req.Properties) == 0 {
 		allConfigs := h.Config.AllConfigs()
-		configs := make(map[string]interface{})
-		for k, v := range allConfigs {
-			configs[k] = v.Value
+		allSettings := h.Config.AllSettings()
+
+		if len(allConfigs) != len(allSettings) {
+			return fmt.Errorf("settings and configs have different length")
+		}
+
+		configs := make(map[string]client.SettingValue)
+		for _, setting := range allSettings {
+			configs[setting.Name] = client.SettingValue{
+				Value:        allConfigs[setting.Name].Value,
+				IsDefault:    allConfigs[setting.Name].IsDefault,
+				DefaultValue: setting.GetDefaultValue(),
+			}
 		}
 		return c.JSON(http.StatusOK, client.GetConfigResult{
 			Configs: configs,
 		})
 	}
 
-	configs := make(map[string]interface{})
+	configs := make(map[string]client.SettingValue)
 	for _, key := range req.Properties {
 		v := h.Config.Get(key)
-		if v.Invalid {
+		s, err := h.Config.GetSetting(key)
+		if v.Invalid || err != nil {
 			continue
 		}
-		configs[key] = v.Value
+		configs[key] = client.SettingValue{
+			Value:        v.Value,
+			IsDefault:    v.IsDefault,
+			DefaultValue: s.GetDefaultValue(),
```

Comment:
I was the one suggesting adding `GetDefaultValue()` rather than exporting the `DefaultValue` field, but now that I look again at the PR, this seems inconsistent with the rest of the code, maybe it's better to export the field instead.

Reply:
yeah on single getter method, i'll revert back to exporting the `DefaultValue`